pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/Comment.java

```diff
@@ -5,45 +5,102 @@
 package net.sourceforge.pmd.lang.java.ast;
 
 import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Map;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
+import org.apache.commons.lang3.StringUtils;
+
+import net.sourceforge.pmd.PMD;
 import net.sourceforge.pmd.lang.ast.AbstractNode;
-import net.sourceforge.pmd.lang.ast.Node;
-import net.sourceforge.pmd.lang.java.javadoc.JavadocTag;
 
 public abstract class Comment extends AbstractNode {
+    // single regex, that captures: the start of a multi-line comment (/**|/*), the start of a single line comment (//)
+    // or the start of line within a multine comment (*). It removes the end of the comment (*/) if existing.
+    private static final Pattern COMMENT_LINE_COMBINED = Pattern.compile("^(?://|/\\*\\*?|\\*)?(.*?)(?:\\*/|/)?$");
 
     protected Comment(Token t) {
         super(-1, t.beginLine, t.endLine, t.beginColumn, t.endColumn);
 
         setImage(t.image);
-        if (t.image.startsWith("/**")) {
-            findJavadocs(t.image);
-        }
     }
 
     @Override
     public String toString() {
         return getImage();
     }
 
-    private void findJavadocs(String commentText) {
+    /**
+     * Filters the comment by removing the leading comment marker (like {@code *}) of each line
+     * as well as the start markers ({@code //}, {@code /*} or {@code /**}
+     * and the end markers (<code>&#x2a;/</code>).
+     * Also leading and trailing empty lines are removed.
+     *
+     * @return the filtered comment
+     */
+    public String getFilteredComment() {
+        List<String> lines = multiLinesIn();
+        lines = trim(lines);
+        return StringUtils.join(lines, PMD.EOL);
+    }
 
-        Collection<JavadocElement> kids = new ArrayList<>();
+    /**
+     * Removes the leading comment marker (like {@code *}) of each line
+     * of the comment as well as the start marker ({@code //}, {@code /*} or {@code /**}
+     * and the end markers (<code>&#x2a;/</code>).
+     *
+     * @param comment the raw comment
+     * @return List of lines of the comments
+     */
+    private List<String> multiLinesIn() {
+        String[] lines = getImage().split("\\R");
```

Comment:
support for the line break matcher is Java8+ only. We can however use the extended version of this:

`\u000D\u000A|[\u000A\u000B\u000C\u000D\u0085\u2028\u2029]`

Reply:
Oh, didn't know that...